packages/plugin-legacy/src/index.ts

```diff
@@ -209,6 +208,11 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
           // See https://github.com/vitejs/vite/pull/10052#issuecomment-1242076461
           overriddenBuildTarget = config.build.target !== undefined
           overriddenDefaultModernTargets = options.modernTargets !== undefined
+
+          // Package is ESM only
+          const { default: browserslistToEsbuild } = await import(
+            'browserslist-to-esbuild'
+          )
```

Comment:
Should dynamic import be placed in conditional judgment? 🤔
```ts
if (options.modernTargets) {
  // Package is ESM only
  const { default: browserslistToEsbuild } = await import(
    'browserslist-to-esbuild'
  )
}
```

Reply:
Good call, I'll change that